tests/cephadm/test_cephadm_pull.py

```diff
@@ -0,0 +1,70 @@
+import ast
+
+import requests
+import yaml
+
+from ceph.ceph_admin import CephAdmin
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def getting_ceph_version(link):
+    """
+    Method return ceph version from magna002 latest ceph container yaml
+    Args:
+        link: magna002 latest ceph container yaml link
+    Returns:
+        ceph version
+    """
+
+    response = requests.get(link)
+    yaml_response = yaml.safe_load(response.text)
+    ceph_version_5 = []
+    for ceph_version in yaml_response:
+        if ceph_version >= 5:
+            ceph_version_5.append(yaml_response[ceph_version]["ceph-version"])
+    return ceph_version_5[-1]
+
+
+def run(ceph_cluster, **kw):
+    """
+    Args: Check the latest image version using cephadm pull option
+        **kw: Key/value pairs of configuration information to be used in the test.
+    Returns:
+        0 - if test case pass
+        1 - it test case fails
+    Test case covered : CEPH-83573787
+    Test Case Flow:
+    1. Deploy a 5.x cluster
+    2. check for container image latest available using cephadm pull option.
+    """
+
+    try:
+        config = kw.get("config")
+        log.info("Executing cephadm pull command")
+        instance = CephAdmin(cluster=ceph_cluster, **config)
+        # Geting ceph version detail using cephadm pull command
+        out, _ = instance.installer.exec_command(cmd="cephadm pull", sudo=True)
+        ceph_version_cephadm = ast.literal_eval(out)["ceph_version"].split(" ")[2][:-6]
+        # Geting ceph version detail from magna002 latest ceph container yaml
+        ceph_version = getting_ceph_version(
+            "http://magna002.ceph.redhat.com/cephci-jenkins/latest-rhceph-container-info/release.yaml"
+        )
+        log.info(
+            "cephadm pulled ceph version: %s\nlatest ceph version available in ceph container yaml: %s"
+            % (ceph_version_cephadm, ceph_version)
+        )
+        # Verfication cephadm pulling latest ceph version image
+        if ceph_version_cephadm == ceph_version:
+            log.info("cephadm pull verification successful")
+            return 0
+        log.info(
+            "cephadm pull verification unsuccessful\ncepadm ceph version: %s\nCeph version available in yaml: %s"
+            % (ceph_version_cephadm, ceph_version)
+        )
+        return 1
+
+    except Exception as e:
```

Comment:
Catching general exception makes this try..catch block useless here. In future, please avoid adding entire code in the try..catch block.

Reply:
Thanks @vamahaja
Removed try and except block